src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -985,5 +990,62 @@ private static IExpressionStatement CreateBoundExpressionStatementOperation(Boun
             return new LazyExpressionStatement(expression, isInvalid, syntax, type, constantValue);
         }
+
+        private static IInterpolatedStringExpression CreateBoundInterpolatedStringExpressionOperation(BoundInterpolatedString boundInterpolatedString)
+        {
+            Lazy<ImmutableArray<IInterpolatedStringContent>> parts = new Lazy<ImmutableArray<IInterpolatedStringContent>>(() => GetInterpolatedStringExpressionParts(boundInterpolatedString));
+            bool isInvalid = boundInterpolatedString.HasErrors;
+            SyntaxNode syntax = boundInterpolatedString.Syntax;
+            ITypeSymbol type = boundInterpolatedString.Type;
+            Optional<object> constantValue = ConvertToOptional(boundInterpolatedString.ConstantValue);
+            return new LazyInterpolatedStringExpression(parts, isInvalid, syntax, type, constantValue);
+        }
+
+        private static IInterpolatedStringContent CreateBoundInterpolatedStringContentOperation(BoundNode boundNode)
+        {
+            if (boundNode == null)
+            {
+                return null;
+            }
+
+            return (IInterpolatedStringContent)s_cache.GetValue(boundNode, n => CreateBoundInterpolatedStringContentCore(n));
+        }
+
+        private static IInterpolatedStringContent CreateBoundInterpolatedStringContentCore(BoundNode boundInterpolatedStringContent)
+        {
+            switch (boundInterpolatedStringContent.Kind)
+            {
+                case BoundKind.StringInsert:
+                    return CreateBoundInterpolationOperation((BoundStringInsert)boundInterpolatedStringContent);
+
+                case BoundKind.Literal:
+                    return CreateBoundInterpolatedStringTextOperation((BoundLiteral)boundInterpolatedStringContent);
+
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(boundInterpolatedStringContent.Kind);
+            }
+        }
+
+        private static IInterpolation CreateBoundInterpolationOperation(BoundStringInsert boundStringInsert)
+        {
+            Lazy<IOperation> expression = new Lazy<IOperation>(() => Create(boundStringInsert.Value));
+            Lazy<IOperation> alignment = new Lazy<IOperation>(() => Create(boundStringInsert.Alignment));
+            Lazy<IOperation> format = new Lazy<IOperation>(() => Create(boundStringInsert.Format));
+            bool isInvalid = boundStringInsert.HasErrors;
+            SyntaxNode syntax = boundStringInsert.Syntax;
+            ITypeSymbol type = null;
+            Optional<object> constantValue = default(Optional<object>);
+            return new LazyInterpolation(expression, alignment, format, isInvalid, syntax, type, constantValue);
+        }
+
+        private static IInterpolatedStringText CreateBoundInterpolatedStringTextOperation(BoundLiteral boundLiteral)
+        {
+            Lazy<IOperation> text = new Lazy<IOperation>(() => CreateInternal(boundLiteral));
```

Comment:
The special handling is needed because BoundLiteralExpression under a BoundInterpolatedStringExpression is special. If we invoke Create method here, then it child of InterpolatedStringText node to be the same InterpolatedStringText node and walker goes into an infinite recursion. Let me stop by to explain it.